flink-tests/src/test/java/org/apache/flink/test/checkpointing/UnalignedCheckpointTestBase.java

```diff
@@ -165,35 +161,11 @@ protected File execute(UnalignedSettings settings) throws Exception {
             miniCluster.after();
         }
         if (settings.generateCheckpoint) {
-            return Files.find(checkpointDir.toPath(), 2, this::isCompletedCheckpoint)
-                    .max(Comparator.comparing(Path::toString))
-                    .map(Path::toFile)
-                    .orElseThrow(() -> new IllegalStateException("Cannot generate checkpoint"));
+            return TestUtils.getMostRecentCompletedCheckpoint(checkpointDir);
```

Comment:
This is not easily possible because I cannot add the `TestUtils` dependency because it would introduce a cycling dependency. Therefore I left it, for now, to not spending more time entangling the dependencies.